File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/recipes/RecipeRetryWithBackOffUntilDeadline.kt
package examples.kotlin.recipes

import com.ing.baker.recipe.kotlindsl.ExperimentalDsl
import com.ing.baker.recipe.kotlindsl.recipe
import kotlin.time.Duration.Companion.hours
import kotlin.time.Duration.Companion.milliseconds
import kotlin.time.Duration.Companion.seconds

@ExperimentalDsl
object RecipeRetryWithBackOffUntilDeadline {
    val recipe = recipe("example") {
        defaultFailureStrategy = retryWithIncrementalBackoff {
            initialDelay = 100.milliseconds
            backoffFactor = 2.0
            maxTimeBetweenRetries = 10.seconds
            until = deadline(24.hours)
        }
    }
}

File: examples/docs-code-snippets/src/main/scala/examples/scala/recipes/RecipeRetryWithBackOffUntilDeadline.scala
package examples.scala.recipes

import com.ing.baker.recipe.common.InteractionFailureStrategy.RetryWithIncrementalBackoff
import com.ing.baker.recipe.common.InteractionFailureStrategy.RetryWithIncrementalBackoff.UntilDeadline
import com.ing.baker.recipe.scaladsl.Recipe

import scala.concurrent.duration.DurationInt

object RecipeRetryWithBackOffUntilDeadline {
  val recipe: Recipe = Recipe("example")
    .withDefaultFailureStrategy(
      RetryWithIncrementalBackoff
        .builder()
        .withInitialDelay(100.milliseconds)
        .withBackoffFactor(2.0)
        .withMaxTimeBetweenRetries(Some(10.minutes))
        .withUntil(Some(UntilDeadline(24.hours)))
        .build()
    )
}